sync/rpc/server.go

```diff
@@ -68,6 +69,43 @@ type PurgeStatus struct {
 	State    types.ProcessState
 }
 
+type CoalesceTask struct {
+	doneChan     chan error
```

Comment:
Don't need a channel. Just create a function to hold a lock and update the number. Can we reuse `PurgeStatus` struct? For example, add a private coalescingProgress field to it for calculation.